apps/frontend/src/app/5_pages/EarnPage/EarnPage.tsx

```diff
@@ -1,12 +1,318 @@
+    if (BigNumber.from(toWei(ZUSDInStabilityPool)).isZero()) {
+      return '0';
+    }
+    return fromWei(
+      BigNumber.from(toWei(poolBalance, 24)).div(toWei(ZUSDInStabilityPool)),
+      4,
+    ).toString();
+  }, [ZUSDInStabilityPool, poolBalance]);
+
+  const isAmountZero = useMemo(() => {
+    return Number(amount) === 0;
+  }, [amount]);
+
+  const newPoolBalance = useMemo(() => {
+    if (isAmountZero) {
+      return 'N/A';
+    }
+    const newBalance = BigNumber.from(poolBalance).add(
+      isDeposit ? amount : -amount,
+    );
+    if (newBalance.lt(0)) {
+      return '0';
+    }
+    return newBalance.toString();
+  }, [poolBalance, isDeposit, amount, isAmountZero]);
+
+  const newPoolBalanceLabel = useMemo(() => {
+    if (isAmountZero) {
+      return 'N/A';
+    }
+    return `${newPoolBalance} ${SupportedTokens.zusd.toUpperCase()}`;
+  }, [isAmountZero, newPoolBalance]);
+
+  const newPoolShare = useMemo(() => {
+    if (isAmountZero) {
+      return 'N/A';
+    }
+
+    const newZUSDInStabilityPool = toWei(ZUSDInStabilityPool).add(
+      toWei(isDeposit ? amount : -amount),
+    );
+
+    if (newZUSDInStabilityPool.isZero()) {
+      return '0 %';
+    }
+    return `${fromWei(
+      BigNumber.from(toWei(newPoolBalance, 24)).div(newZUSDInStabilityPool),
+      4,
+    ).toString()} %`;
+  }, [ZUSDInStabilityPool, amount, isAmountZero, isDeposit, newPoolBalance]);
+
+  const maximumAmount = useMemo(() => {
+    if (isDeposit) {
+      return balance;
+    } else {
+      return poolBalance;
+    }
+  }, [balance, isDeposit, poolBalance]);
+
+  const isValidAmount = useMemo(
+    () => Number(amount) <= Number(maximumAmount),
+    [amount, maximumAmount],
+  );
+
+  const isSubmitDisabled = useMemo(
+    () => !account || !amount || Number(amount) <= 0 || !isValidAmount,
+    [account, amount, isValidAmount],
+  );
+
+  const onTransactionSuccess = useCallback(() => {
+    getStabilityDeposit();
+    getZUSDInStabilityPool();
+  }, [getStabilityDeposit, getZUSDInStabilityPool]);
+
+  return (
+    <div className="w-full flex flex-col items-center text-gray-10 mt-9 sm:mt-24">
+      <Heading className="text-base sm:text-2xl">
+        {t(pageTranslations.title)}
+      </Heading>
+      <Paragraph
+        size={ParagraphSize.base}
+        className="mt-2.5 sm:mt-4 sm:text-base"
+      >
+        {t(pageTranslations.subtitle)}
+      </Paragraph>
+
+      <div className="mt-12 w-full p-0 sm:border sm:border-gray-50 sm:rounded sm:w-[28rem] sm:p-6 sm:bg-gray-90">
+        <div className="w-full flex flex-row justify-between items-center mb-4">
+          <Tabs
+            size={TabSize.small}
+            type={TabType.secondary}
+            items={actions}
+            onChange={setIndex}
+            index={index}
+          />
+
+          <button
+            onClick={onMaximumAmountClick}
+            className="text-xs font-medium underline whitespace-nowrap"
+            {...applyDataAttr('max-button')}
```

Comment:
```suggestion
            {...applyDataAttr('earn-max-button')}
```

Prefix all the data attributes on this page with `earn-`